extension/app/ts/background/windows/interceptorAccess.ts

```diff
@@ -1,44 +1,67 @@
 import { addressString } from '../../utils/bigint.js'
 import { Future } from '../../utils/future.js'
 import { imageToUri } from '../../utils/imageToUri.js'
-import { AddressInfoEntry, PendingAccessRequest } from '../../utils/user-interface-types.js'
+import { PopupMessage } from '../../utils/interceptor-messages.js'
+import { AddressInfoEntry, PendingAccessRequestArray } from '../../utils/user-interface-types.js'
 import { setAccess, updateWebsiteApprovalAccesses } from '../accessManagement.js'
 import { sendPopupMessageToOpenWindows } from '../backgroundUtils.js'
 import { updateExtensionBadge } from '../iconHandler.js'
 import { findAddressInfo } from '../metadataUtils.js'
-import { savePendingAccessRequests, saveWebsiteAccess, WebsiteAccess } from '../settings.js'
+import { savePendingAccessRequests, saveWebsiteAccess, WebsiteAccessArray } from '../settings.js'
 
-export type Confirmation = 'Approved' | 'Rejected' | 'NoResponse'
-let pendingInterceptorAccess: Future<Confirmation> | undefined = undefined
+export type Confirmation = {
+	outcome: 'Approved' | 'Rejected' | 'NoResponse',
+	origin : string,
+	requestAccessToAddress: bigint | undefined,
+}
 
 let openedInterceptorAccessWindow: browser.windows.Window | null = null
 
+let pendingInterceptorAccess: {
+	future: Future<Confirmation>
+	origin: string,
+	requestAccessToAddress: bigint | undefined,
+} | undefined = undefined
+
 const onCloseWindow = () => { // check if user has closed the window on their own, if so, reject signature
-	if (pendingInterceptorAccess !== undefined) pendingInterceptorAccess.resolve('NoResponse')
+	if (pendingInterceptorAccess !== undefined) pendingInterceptorAccess.future.resolve({
+		outcome: 'NoResponse',
+		origin: pendingInterceptorAccess.origin,
+		requestAccessToAddress: pendingInterceptorAccess.requestAccessToAddress
+	})
 	pendingInterceptorAccess = undefined
-	window.interceptor.interceptorAccessDialog = undefined
 	openedInterceptorAccessWindow = null
 	browser.windows.onRemoved.removeListener( onCloseWindow )
 }
 
+export async function resolveExistingInterceptorAccessAsNoResponse() {
+	if (pendingInterceptorAccess === undefined) return
+	await resolveInterceptorAccess({
+		outcome: 'NoResponse',
+		origin: pendingInterceptorAccess.origin,
+		requestAccessToAddress: pendingInterceptorAccess.requestAccessToAddress
+	})
+}
+
 export async function resolveInterceptorAccess(confirmation: Confirmation) {
-	if (pendingInterceptorAccess !== undefined) pendingInterceptorAccess.resolve(confirmation)
+	if (pendingInterceptorAccess !== undefined)
+	if (confirmation.origin === pendingInterceptorAccess.origin && confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress)
+	pendingInterceptorAccess.future.resolve(confirmation)
```

Comment:
```suggestion
	if (pendingInterceptorAccess !== undefined) {
		if (confirmation.origin === pendingInterceptorAccess.origin && confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress) {
			pendingInterceptorAccess.future.resolve(confirmation)
		}
	}
```
```suggestion
	if (pendingInterceptorAccess !== undefined
		&& confirmation.origin === pendingInterceptorAccess.origin
		&& confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress
	) {
		pendingInterceptorAccess.future.resolve(confirmation)
	}
```
```suggestion
	if (pendingInterceptorAccess !== undefined && confirmation.origin === pendingInterceptorAccess.origin && confirmation.requestAccessToAddress === pendingInterceptorAccess.requestAccessToAddress) pendingInterceptorAccess.future.resolve(confirmation)
```

Reply:
these should be guard clauses, I am clearly been too tired when I wrote this, the first line is also a bug :(